src/index.js

```diff
@@ -21,12 +21,37 @@ function* watcherSaga() {
 
 
 //reducers go here
-const gifReducer = (state = '', action) => {
-    switch (action.type === 'GET_GIF') {
-        
-    }
-}
+const searchReducer = (state = '', action) => {
+    switch (action.type) {
+        case "GET_GIF":
+            console.log('Getting GIF', action.payload.data);
+            let searchResults = [];
+            let results = action.payload.data;
+            for (let gif of results) {
+                console.log(gif);
+                searchResults.push({url: gif.url, id: gif.id});
+            }
+            return searchResults;
+        default:
+            return state;
+    };
+};
+
+const favoriteReducer = (state = [], action) => {
+    switch (action.type) {
+        case "ADD_FAVORITE":
+            console.log(`Trying to add ${action.payload} to favorites`);
+            state.push(action.payload);
```

Comment:
We're supposed to use spread instead of push in reducers, otherwise we're mutating state.